src/grouptableaux.jl

```diff
@@ -0,0 +1,284 @@
+#using QuantumClifford
+#= using GLMakie
+using GraphMakie =#
+using Graphs
+using LinearAlgebra
+#using IterTools
+
+
+#
+
+# center
+# normalizer
+# `invert' arbitrary groups
+
+"""
+input: (generating) set of PauliOperator
+output: Tableau
+```jldoctest
+julia> groupify(S"XZ XZ")
++ __
++ XZ
++ ZX
++ YY
+```
+"""
+function groupify(s::Stabilizer) # add checks?
+    n = length(s)
+    group = zero(Stabilizer, 2^n, nqubits(s))
+    for i in 0:2^n-1
+        for (digit_order, j) in enumerate(digits(i, base=2, pad=n))
+            if j == 1
+                group[i+1] *= s[digit_order]
+            end
+        end
+    end
+    return group # TODO use Tableau instead of Stabilizer
+end
+
+
+"""
+For a given non-necessarily-minimal generating set represented as a tableau `s`,
+return the minimal generating set.
+"""
+function get_generating_set(s) # TODO potentially rename to `generator_tableau` or `generating_set`
+    s, _, r = canonicalize!(copy(s), ranks=true)
+    if r == 0
+        gs = zero(Stabilizer, 1, nqubits(s))
+        if (1im * zero(Stabilizer, 1, nqubits(s))[1]) in s
+            gs[1] = 1im * gs[1]
+        elseif (-1 * zero(Stabilizer, 1, nqubits(s))[1]) in s
+            gs[1] = -1 * gs[1]
+        end
+        return gs
+    else
+        return s[1:r, :]
+    end
+end
+
+"""
+For a set of logical operators, rewrite in a set of logical x and logical z with the appropriate pairwise anticommutation
+""" # TODO unfinished, including docs
+#= function logical_operator_canonicalize(s)
+    # TODO potentially use
+    # julia> MixedDestabilizer(S"XXX") |> logicalxview
+    # julia> MixedDestabilizer(S"XXX") |> logicalzview
+    # or stabilizerview or destabilizerview
+    # @assert length(s) == length(get_generating_set(groupify(s)))
+    mx = MixedDestabilizer(s)
+    return vcat(logicalxview(mx), logicalzview(mx), s), rank(mx) # TODO use Tableau instead of Stabilizer
+end =#
+
+
+##
+# s = S"XZ ZX IY"
+# s = S"ZZ XZ YY"
+# println("")
+# t = logical_operator_canonicalize(s)
+# ##
+
+#= function commutavise(s::Stabilizer) # TODO take a Tableau
+    @assert length(s) == length(groupify(s))
+    embedding = []
+    # currently this works only on generating sets for abelian groups
+    # maybe split in commutavise
+    # news = ... with n+(n-rank) columns
+    # news[:, 1:n] = s
+    # add some value to news[:, n+1:2n-rank]
+    for P in s
+        for Q in s
+            if comm(P, Q) == 0x01
+                # finish
+            end
+        end
+    end
+end =#
+
+# s = S"XZ ZX IY"
+# s = groupify(s)
+# println("")
+# print(get_generating_set(s))
+# ##
+"""
+Returns the full pauli group of the given length. Phases besides + are ignored by default, but can be included by setting phases = true. 
+"""
+function pauligroup(n, phases=false)
```

Comment:
```suggestion
function pauligroup(n; phases=false)
```

That way it is a keyword argument instead of a default argument (makes things a bit more legible)